storm-core/src/jvm/backtype/storm/utils/StormCommandExecutor.java

```diff
@@ -0,0 +1,868 @@
+                String[] tokens = line.split(" ");
+                if ("VALUE:".equals(tokens[0])) {
+                    confValue = StringUtils.join(Arrays.copyOfRange(tokens, 1,
+                            tokens.length), " ");
+                    break;
+                }
+            }
+            br.close();
+        } catch (Exception ex) {
+            System.out.println("Exception occured while starting process via " +
+                    "processbuilder " + ex.getMessage());
+        }
+        return confValue;
+    }
+
+    void executeStormClass (String className, String jvmType, List<String>
+            jvmOptions, List<String> extraJars, List<String> args, boolean
+            fork, boolean daemon, String daemonName) {
+        List<String> extraPaths = new ArrayList<>();
+        extraPaths.add(this.clusterConfDirectory);
+        String stormLogDirectory = this.confValue("storm.log.dir",
+                extraPaths, daemon);
+        if ((stormLogDirectory == null) || ("".equals(stormLogDirectory)) ||
+                ("nil".equals(stormLogDirectory))) {
+            stormLogDirectory = this.stormHomeDirectory + this.fileSeparator
+                    + "logs";
+        }
+        List<String> commandList = new ArrayList<String>();
+        commandList.add(this.javaCommand);
+        commandList.add(jvmType);
+        commandList.add("-Ddaemon.name=" + daemonName);
+        commandList.add(this.getConfigOptions());
+        commandList.add("-Dstorm.home=" + this.stormHomeDirectory);
+        commandList.add("-Dstorm.log.dir=" + stormLogDirectory);
+        commandList.add("-Djava.library.path=" + this
+                .confValue("java.library.path", extraJars, daemon));
+        commandList.add("-Dstorm.conf.file=" + this.configFile);
+        commandList.add("-cp");
+        commandList.add(this.getClassPath(extraJars, daemon));
+        commandList.addAll(jvmOptions);
+        commandList.add(className);
+        commandList.addAll(args);
+        ProcessBuilder processBuilder = new ProcessBuilder(commandList);
+        processBuilder.inheritIO();
+        try {
+            Process process = processBuilder.start();
+            System.out.println("Executing the command: ");
+            String commandLine = StringUtils.join(commandList, " ");
+            System.out.println(commandLine);
+            if (daemon == true) {
+                Runtime.getRuntime().addShutdownHook(new ShutdownHookThread
+                        (process, commandLine));
+            }
+            System.out.println("Waiting for subprocess to finish");
+            process.waitFor();
+            System.out.println("subprocess finished");
+            System.out.println("Exit value from subprocess is :" + process
+                    .exitValue());
+        } catch (Exception ex) {
+            System.out.println("Exception occured while starting process via " +
+                    "processbuilder " + ex.getMessage());
+        }
```

Comment:
not all process need waitfor(). maybe you can add option. Like this:

    public static java.lang.Process launch_process(final String command, final Map<String, String> environment, boolean backend) throws IOException {

        if (backend == true) {
            new Thread(new Runnable() {

                @Override
                public void run() {
                    String[] cmdlist = (new String("nohup " + command + " &")).split(" ");
                    try {
                        launchProcess(cmdlist, environment);
                    } catch (IOException e) {
                        LOG.error("Failed to run " + command + ":" + e.getCause(), e);
                    }
                }
            }).start();
            return null;
        } else {
            String[] cmdlist = command.split(" ");
            return launchProcess(cmdlist, environment);
        }
    }